nano/secure/common.cpp

```diff
@@ -738,6 +754,14 @@ bool nano::vote::validate () const
 	return nano::validate_message (account, hash (), signature);
 }
 
+uint64_t nano::vote::packed_timestamp (uint64_t timestamp, uint8_t duration) const
+{
+	debug_assert ((timestamp & 0xf) == 0 && "Invalid timestamp");
+	debug_assert ((duration & 0xf0) == 0 && "Invalid duration");
```

Comment:
Since you have `duration_max`, it would be slightly better to compare against it here.